components/AnalysisSummary.tsx

```diff
@@ -0,0 +1,81 @@
+import { Analysis, AnalysisStrategy, ExperimentFull } from '@/models'
+import _ from 'lodash'
```

Comment:
Because we only use a small fraction of lodash, it might be better to import just what we need.

```suggestion
import filter from 'lodash/filter'
import groupBy from 'lodash/groupBy'
import orderBy from 'lodash/orderBy'
import sortBy from 'lodash/sortBy'
```

There is a babel plugin we can add to do this for us.

There is also Webpack's tree-shacking. I am not sure if Next.js has it on my default.

We can run `npm run analyze` to see what is being pulled into the bundle. If it is there, then we should address it at some point. Could wait until a later optimization phase.

Okay, just finished looking the output of `npm run analyze`. We'll should do something with MUI too. Whole library being bundled.

Reply:
> There is a babel plugin we can add to do this for us.

So we would import `_` and it'll break it up as part of the build? I'm happy to add specific imports, but using `_` is easier when developing, and I like the namespacing.

> We can run `npm run analyze` to see what is being pulled into the bundle. If it is there, then we should address it at some point. Could wait until a later optimization phase.
> 
> Okay, just finished looking the output of `npm run analyze`. We'll should do something with MUI too. Whole library being bundled.

Yeah, let's open a separate issue for that. It'd be good to do once things are more stable in terms of what we have on the pages.

Reply:
The one I know of is https://www.npmjs.com/package/babel-plugin-transform-imports. Just took a peak. May not work with just the `_` namespace.
There are a lot of tools to address the over-bundling issue. It is definitely a pre-optimization. Opening an issue as a reminder.

Reply:
FWIW, it's [not recommended to pull in the entirety of lodash](https://www.blazemeter.com/blog/the-correct-way-to-import-lodash-libraries-a-benchmark) unless it really makes sense. That blog post recommends https://github.com/lodash/babel-plugin-lodash to address this so you can just use `_.map()` instead of `map()` but it goes into why you want to import one-by-one instead of everything.

The biggest downside for using a plugin like this is that it can greatly increase build times. IMHO, it's easier to just import one by one (and it's what is done in Calypso fwiw).

Reply:
Another solution I used in a previous project was to create a module where different utilities are imported and then default exported in an object. I did this specifically for `lodash` but it could be a general solution.

Below is basically what I did:

```
// utils.ts
import filter from 'lodash/filter'
import map from 'lodash/map'

export default {
  filter,
  map,
}
```

```
import _ from '@/utils'

_.map(...)
_.filter(...)
```
Or
```
import utils from '@/utils'

utils.map(...)
utils.filter(...)
```
So there is no confusion if we import/export from other libraries besides lodash.

BTW, the entire lodash library is a small fraction compared to the entire MUI library.

Reply:
> The biggest downside for using a plugin like this is that it can greatly increase build times.

I'd be curious how much some of these plugins increase build times. For plugins that simply do optimizations, they can be configured to only run for a production build to help keep development fast(er).

Reply:
> I'd be curious how much some of these plugins increase build times.

I imagine it's largely related to the application size and it's use of the plugin. If it only has to change a few files, it's probably negligible. If it were used in something like Calypso... :shudder: Calypso already takes long enough to build :D

Reply:
> BTW, the entire lodash library is a small fraction compared to the entire MUI library.

:arrow_up: This.

I don't mind using individual imports to resolve this. The babel plugin hasn't been updated in ages. However, let's not get distracted by micro-optimisations and remember our likely user base for the foreseeable future: Repeat visitors on a good connection with populated cache. We can afford larger bundle sizes if they make life easier, especially at this point. :slightly_smiling_face:

Reply:
Agreed. We have low-priority #104 to remind us later to optimize bundle if it ever becomes a "real" issue.